pepper-apis/src/main/java/org/broadinstitute/ddp/model/event/ActivityInstanceCreationEventAction.java

```diff
@@ -167,6 +168,41 @@ public void doAction(Handle handle, EventSignal signal) {
                 studyActivityId,
                 signal.getStudyId(),
                 InstanceStatusType.CREATED));
+
+        // Create child nested activity instances, if any.
+        List<Long> childActIdsToCreate = handle.attach(JdbiActivity.class)
+                .findChildActivityIdsThatNeedCreation(studyActivityId);
+        for (var childActivityId : childActIdsToCreate) {
+            String childInstanceGuid = jdbiActivityInstance.generateUniqueGuid();
+            long newChildInstanceId = jdbiActivityInstance.insert(
+                    childActivityId,
+                    signal.getParticipantId(),
+                    childInstanceGuid,
+                    null,
+                    Instant.now().toEpochMilli(),
+                    null,
+                    newActivityInstanceId,
+                    null,
+                    null,
+                    null
+            );
+            jdbiActivityInstanceStatus.insert(
+                    newChildInstanceId,
+                    InstanceStatusType.CREATED,
+                    Instant.now().toEpochMilli(),
+                    signal.getParticipantId()
+            );
+            LOG.info("Created child instance {} for parent instance {}", childInstanceGuid, activityInstanceGuid);
+            EventService.getInstance().processAllActionsForEventSignal(handle, new ActivityInstanceStatusChangeSignal(
+                    signal.getOperatorId(),
+                    signal.getParticipantId(),
+                    signal.getParticipantGuid(),
+                    signal.getOperatorGuid(),
+                    newChildInstanceId,
+                    childActivityId,
+                    signal.getStudyId(),
+                    InstanceStatusType.CREATED));
+        }
```

Comment:
I could have **sworn** that I had seen the child instance creation logic in your original PRs!

Reply:
I thought I had covered it too. But realized that the creation event uses "lower-level" facilities to create instances so it missed out on handling child activities.